Add support for featured and high quality apps (#677)
YunoHost/apps: README.md

> ./add_or_update.py [community.json OR official.json] [github/gitlab url OR app name [github/gitlab url OR app name [github/gitlab url OR app name ...]]]

Reviewer: no more `community.json OR official.json`

Author: This PR will not remove Official.json or Community.json (not yet at least...)

Reviewer: Sure, but when the new list will be there, a packager who want to add (or update) an app, have to do it on the new list, and tools will regenerate old ones (there are some code to do for that).

Author: A way or another, we would rather use apps.json as the main list. So I guess we should indeed modify this script as well.

Author: Done here https://github.com/YunoHost/apps/pull/694/commits/1c7c403dd048ae906f8458f380897f88f52ab2eb, for the PR https://github.com/YunoHost/apps/pull/694.
The current PR will not create the apps.json list yet.

```

### How to make my app a High Quality app ?

> A High Quality app will be highlighted in the app list and mark as a level 8 app.  

Reviewer 2: ```suggestion
A High Quality app will be highlighted in the app list and marked as a level 8 app.  
```

> To become a High Quality app, a package have to follow the following rules:

Reviewer 2: ```suggestion
To become a High Quality app, a package has to follow the following rules:
```

> * The app should already be in the community list for 2 months.

Reviewer 2: ```suggestion
* The app should already have been in the community list for 2 months.
```

> * The app should be keep up to date, regarding the upstream source. (If it’s possible with our current YunoHost version)

Reviewer 2: ```suggestion
* The app should be kept up to date, regarding the upstream source (if it’s possible with our current YunoHost version).
```

* The package itself should be up to date regarding the packaging recommendations and helpers.

> * The package should be level 7, at least.

Reviewer 3: Shouldn't we also add the requirement for X months / weeks here ?

Author: Yep I guess

* The repository should have testing and master branches, at least. The list should point to HEAD, so the list stays up to date.

> * Any modification should be done to the testing branch, and wait at least for one approval for one member of the Apps group. So that we can ensure that there’s nothing in opposition to those criteria. Nor any changes that would harm servers.

Reviewer 2: ```suggestion
* Any modification should be done to the testing branch, and wait at least for one approval of one member of the Apps group so that we can ensure that there’s nothing in opposition to those criteria, nor any changes that would harm servers.
```

> * The package should comply with the [requirement of the level 8](https://github.com/YunoHost/doc/blob/master/packaging_apps_levels.md#level-8).

Reviewer 2: ```suggestion
* The package should comply with the [requirements of the level 8](https://github.com/YunoHost/doc/blob/master/packaging_apps_levels.md#level-8).
```

You can find the validation form used by Apps group [here](https://github.com/YunoHost/apps/blob/master/hq_validation_template.md).

> If the app is already tag as High Quality and one of those criteria isn't respected anymore. After a warning, the tag will be removed until the criteria are again validated.

Reviewer 2: ```suggestion
If the app is already tagged as High Quality and one of those criteria isn't respected anymore: after a warning, the tag will be removed until the criteria is again validated.
```

Author: "criteria" is plural.

Reviewer 2: your sentence refers to "one of those criteria"...

Author: It would be "criterion is" then. but I don't know how to commit into your suggestion

Reviewer 2: suggestion edited (trying to get the hang of it...)

To make an app a High Quality app, technically, you have to add the tag ```"high_quality": true```.
```json
    "wallabag": {
        "branch": "master",
        "high_quality": true,
        "revision": HEAD,
        "url": "https://github.com/abeudin/wallabag_ynh.git",
        "state": "working"
    }
```

### How to make my app a Featured app ?

> A Featured app highlighted in the app list and shown before any others.  

Reviewer 2: ```suggestion
A Featured app is highlighted in the app list and shown before any others.  
```

> To become a Featured app, a package have to follow the following rules:

Reviewer 2: ```suggestion
To become a Featured app, a package has to follow the following rules:
```

* The app should already be a High Quality app.
* The upstream app should be accessible and well made.
* The app should be interesting and demanded by the community.
* The app should fit the spirit of YunoHost.

> **Please note that the exact process to decide which app are going to be Featured, and for how many time, isn't yet defined...**

Reviewer 2: ```suggestion
**Please note that the exact process to decide which apps are going to be Featured, and for how many time, isn't yet defined...**
```

> To make an app a Featured app, technically, you have to add the tag ```"featured": true```.

Reviewer 2: Already stated before

Author: Really ?

Reviewer 2: Oops :relaxed:

```json
    "wallabag": {
        "branch": "master",
        "high_quality": true,
        "featured": true,
        "revision": HEAD,
        "url": "https://github.com/abeudin/wallabag_ynh.git",
        "state": "working"
    }
```

### What to do if I can't maintain my app anymore ?

If you don't have time anymore to maintain an app, you can update its status to inform users and packagers that you will not maintain it anymore.  
In order to do so, use the tag `"maintained":`.  
This tag can have 5 different values:
- `"maintained": true` That's the default value if the tag isn't present for your app. That simply means that this app is maintained.
- `"maintained": "request_help"` Use that value to inform other packagers that you need help to maintain this app. You'll then be more than one maintainer for this apps.
- `"maintained": "request_adoption"` Use that value to inform other packagers, as well as users, that you're going to give up that app. So that you would like another maintainer to take care of it.

> - `"maintained": false` or `"maintained": "orphaned"` This value means that this app is no longer maintained... That means also that a packager can declare himself as its new maintainer.  

Reviewer 2: ```suggestion
- `"maintained": false` or `"maintained": "orphaned"` This value means that this app is no longer maintained... That means also that a packager can declare himself/herself as its new maintainer.  
```

Author: ...
Hope we're not going to start to do that every where in English as well...

> Please contact Apps group if you want to take back a unmaintained app.

Reviewer 2: ```suggestion
Please contact Apps group if you want to take back an unmaintained app.
```

> Please contact Apps group if you want to take back an unmaintained app.

Reviewer 3: ```suggestion
Please contact the Apps group if you want to take care of an unmaintained app.
```

> If you want to modify the status of one of your app, for any reason, please think also to inform the community via the forum. Users would probably be glad to be inform that an app they use will be unmaintained.

Reviewer 2: ```suggestion
If you want to modify the status of one of your apps, for any reason, please think also to inform the community via the forum. Users would probably be glad to be informed that an app they use will become unmaintained.
```

> If you want to modify the status of one of your apps, for any reason, please think also to inform the community via the forum. Users would probably be glad to be informed that an app they use will become unmaintained.

Reviewer 3: ```suggestion
If you want to modify the status of one of your apps, for any reason, please consider informing the community via the forum. Users would probably be glad to be informed that an app they use will become unmaintained.
```

If the app is already tag as High Quality and one of those criteria isn't respected anymore. After a warning, the tag will be removed until the criteria are again validated.

To make an app a High Quality app, technically, you have to add the tag ```"high_quality": true```.
```json
    "wallabag": {
        "branch": "master",
        "high_quality": true,
        "revision": HEAD,
        "url": "https://github.com/abeudin/wallabag_ynh.git",
        "state": "working"
    }
```

### How to make my app a Featured app ?

A Featured app highlighted in the app list and shown before any others.  
To become a Featured app, a package have to follow the following rules:

* The app should already be a High Quality app.
* The upstream app should be accessible and well made.
* The app should be interesting and demanded by the community.
* The app should fit the spirit of YunoHost.

**Please note that the exact process to decide which app are going to be Featured, and for how many time, isn't yet defined...**

> To make an app a High Quality app, technically, you have to add the tag ```"featured": true```.

Reviewer: Sorry, I didn't point this specific line. I think it's Featured and not High Quality

Author: Oh yes indeed, forgot this one
